tracer/src/Datadog.Trace/SpanContext.cs

```diff
@@ -199,7 +199,7 @@ IEnumerator IEnumerable.GetEnumerator()
                     return true;
 
                 case HttpHeaderNames.SamplingPriority:
-                    var samplingPriority = SamplingPriority ?? TraceContext?.SamplingPriority;
```

Comment:
This code is only used to communicate the spancontext from one version of the tracer to the other. `?? TraceContext?.SamplingPriority` was originally needed because of the reasons you mention, but with this PR the sampling priority at the trace level is synchronized by other means.

Reply:
Ok, thanks for clarifying.